apps/system/js/app_chrome.js

```diff
@@ -604,10 +637,34 @@
     });
   };
 
+  AppChrome.prototype._onBookmarksChange = function (evt) {
+    var targetUrl = evt.target.id;
+    if (this._currentURL === targetUrl) {
```

Comment:
Why this check? We can unpin sites from a different URL than the pinned one, isn't it?